verifcid/allowlist.go

```diff
@@ -0,0 +1,58 @@
+package verifcid
+
+import (
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// Allowlist defines an interface containing list of allowed multihashes.
+type Allowlist interface {
+	// IsAllowed checks for multihash allowance by the code.
+	IsAllowed(code uint64) bool
+	// ValidateCid validates multihash allowance behind given CID.
+	ValidateCid(c cid.Cid) error
+}
+
+// NewAllowlist constructs new Allowlist from the given map set.
+func NewAllowlist(allowset map[uint64]bool) Allowlist {
+	return &allowlist{allowset: allowset}
+}
+
+type allowlist struct {
+	allowset map[uint64]bool
+}
+
+func (ghr *allowlist) IsAllowed(code uint64) bool {
+	good, found := ghr.allowset[code]
+	if good {
+		return true
+	}
+
+	if !found {
+		if code >= mh.BLAKE2B_MIN+19 && code <= mh.BLAKE2B_MAX {
+			return true
+		}
+		if code >= mh.BLAKE2S_MIN+19 && code <= mh.BLAKE2S_MAX {
+			return true
+		}
```

Comment:
This is weird, the allow list takes in a map of allowed codecs, but have special case for blake hardcoded ?

I don't personally mind if a whole bunch of new cases are added to the map instead, this isn't high performance code.

Reply:
`mh.BLAKE2B_MAX-mh.BLAKE2B_MIN+19=82`

I don't think it's a good idea to type in all 82 cases manually. Alternatively, this could be a loop filling up the goodset, but then it's not any better than current ifs.

Reply:
Ok, so you kept ifs, nice